orttraining/orttraining/python/training/onnxblock/blocks.py

```diff
@@ -28,8 +29,9 @@ class Block(ABC):
         base (onnx.ModelProto): The base model that the subclass can manipulate.
     """
 
-    def __init__(self):
+    def __init__(self, temp_file_name="temp.onnx"):
         self.base = None
+        self.temp_onnx_path = temp_file_name
```

Comment:
Does the temp_file_name need to be known in the constructor? Can we just use `tempfile` python module and create a temporary directory on the go?